### PR TITLE
[SDK-1407] Add Branch QR code methods

### PR DESCRIPTION
**Pull request by nsingh-branch**
### Added a new function, `getBranchQRCode()`, to generate Branch QR codes.
- getBranchQRCode()
    - Takes 4 arguments: QR code settings, universal object,  link properties, and link control parameters.
    - Returns a base64 encoded string of the generated QR code image
 - Also updated the Android SDK to v5.2.0 and the iOS SDK to v1.43.1
    
### Sample Code
```
  function BranchQRCode() {
    var qrCodeSettings = {
        width: 2000,
        codeColor: "#3b2016",
        backgroundColor: "#a8e689",
        centerLogo: "https://cdn.britannica.com/95/156695-131-FF89C9FA/oak-tree.jpg",
        margin: 4,
        imageFormat: "PNG"
    };

    var analytics = {
        channel: "facebook",
        feature: "onboarding",
        campaign: "content 123 launch",
        stage: "new user",
        tags: ["one", "two", "three"],
        alias: document.getElementById("alias").value
      };
    
    var properties = {
        $desktop_url: "http://www.example.com/desktop",
        $android_url: "http://www.example.com/android",
        $ios_url: "http://www.example.com/ios",
        $ipad_url: "http://www.example.com/ipad",
        $deeplink_path: "content/123",
        $match_duration: 2000,
        custom_string: "data",
        custom_integer: Date.now(),
        custom_boolean: true
    };
    
    
    if (branchUniversalObj === null) {
        alert("Need to Generate Branch Universal Object");
        return;
    }
      
    Branch
    .getBranchQRCode(qrCodeSettings, branchUniversalObj, analytics, properties)
    .then(function success(res) {
        document.getElementById("qrCodeImage").src = "data:image/png;base64, " + res;
    })
    .catch(function error(err) {
        alert(err, true);
    });
  }
```

### CODE REVIEW

File: src/android/io/branch/BranchSDK.java [hunk 1]
```diff
@@ -508,6 +519,60 @@ private void generateShortUrl(int instanceIdx, JSONObject options, JSONObject co
...
+     */
+    private void getBranchQRCode(JSONObject qrCodeSettings, int instanceIdx, JSONObject options, JSONObject controlParams, CallbackContext callbackContext) throws JSONException {
+
+        BranchLinkProperties linkProperties = createLinkProperties(options, controlParams);
+
+        BranchUniversalObjectWrapper branchUniversalWrapper = (BranchUniversalObjectWrapper) this.branchObjectWrappers.get(instanceIdx);
+        BranchUniversalObject buo = branchUniversalWrapper.branchUniversalObj;
+
+        BranchQRCode branchQRCode = new BranchQRCode();
+        if (qrCodeSettings.has("codeColor")) branchQRCode.setCodeColor(qrCodeSettings.getString("codeColor"));
```
Comment (gdeluna-branch) [inline src/android/io/branch/BranchSDK.java:538]: Code style: if statement bodies should always have a new line and be encased in braces. An example of this style is under `createLinkProperties`

File: cordova-ionic-phonegap-branch-deep-linking-attribution.iml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (gdeluna-branch) [inline cordova-ionic-phonegap-branch-deep-linking-attribution.iml:1]: This is an auto generated file from your environment- I recommend excluding this change then adding *.iml to the .gitignore

File: src/index.js [hunk 3]
```diff
@@ -314,5 +314,29 @@ Branch.prototype.lastAttributedTouchData = function lastAttributedTouchData() {
   return execute("lastAttributedTouchData");
 };
 
+Branch.prototype.getBranchQRCode = function getBranchQRCode(
+  qrCodeSettings,
+  branchUniversalObject,
+  analytics,
+  properties
+) {
+  var args = [];
```
Comment (gdeluna-branch) [inline src/index.js:323]: Does this need to be wrapped in a Promise? There is a network request being made here.
cc @JagadeeshKaricherla-branch

**Review by gdeluna-branch — CHANGES_REQUESTED** [review]
Awaiting resolution on the Promise

**Review by gdeluna-branch — APPROVED** [review]

**Review by JagadeeshKaricherla-branch — APPROVED** [review]

**Pull request merged**